### PR TITLE
Single Executable for main GDAS JEDI applications

### PR DESCRIPTION
**Pull request by danholdaway**
# Description

Changes that accompany GDAS PR (NOAA-EMC/GDASApp/pull/1075) that allows building of a single gdas executable, which should be more compliant with NCO requirements.

Addresses https://github.com/NOAA-EMC/GDASApp/issues/1085

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- GDAS ctests on Hera.

# Checklist
- [ ] Any dependent changes have been merged and published
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [X] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: sorc/link_workflow.sh
Comment (aerorahul) [inline sorc/link_workflow.sh:336]: Probably need to add/remove these to the `.gitignore`

**Review by aerorahul — APPROVED** [review]
looks good. just one minor comment.

File: sorc/link_workflow.sh
Comment (danholdaway) [reply sorc/link_workflow.sh:336]: [Looks like](https://github.com/danholdaway/global-workflow/blob/264f2fa2cbf2e6438113a8258c1468656c471a40/.gitignore#L20) it takes care of the whole directory in one go.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests

**Comment by aerorahul** [comment]
I think the GDASApp tests are expected to fail as there is a PR in review there that this PR needs.
We will update the PR template in the repo to make the dependency clear.
However, the failure in the AOWCDA test is not expected.  Something else must be missing.

**Comment by WalterKolczynski-NOAA** [comment]
Marking as draft until the dependent PR is merged and this PR updated with the new hash.

**Comment by danholdaway** [comment]
> I think the GDASApp tests are expected to fail as there is a PR in review there that this PR needs. We will update the PR template in the repo to make the dependency clear. However, the failure in the AOWCDA test is not expected. Something else must be missing.

That error is:

```
FileNotFoundError: [Errno 2] No such file or directory: '/scratch1/NCEPDEV/global/CI/2565/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_264f2fa2/gdas.20210324/12//model_data/ice/restart/20210324.180000.cice_model.res.nc'
```

which I don't think seems related to the changing of the executables. Do you have some suggestions for how to run this and debug it?

**Comment by danholdaway** [comment]
@WalterKolczynski-NOAA I removed the draft tag since the GDAS PR was merged.

**Comment by RussTreadon-NOAA** [comment]
Do we plan on getting g-w PR #2477 into `develop` before this PR?

**Comment by danholdaway** [comment]
@RussTreadon-NOAA I think this one should go first, since the GDASapp PR was merged.

**Comment by RussTreadon-NOAA** [comment]
OK.  I'll run JEDI ATM CI on Hera from this PR and see what happens.

**Comment by danholdaway** [comment]
@RussTreadon-NOAA before you do that perhaps we try to resolve the issue of where to send the ensemble. Reviewing your comment, I think it might be preferable to fix via another change in GDAS instead of the change in the GW.

**Comment by RussTreadon-NOAA** [comment]
Agreed.  Chasing multiple PRs is a problematic.  I'll first focus on g-w PR #2477 and GDASApp PR [#1033](https://github.com/NOAA-EMC/GDASApp/pull/1033)

**Comment by danholdaway** [comment]
@aerorahul @WalterKolczynski-NOAA I fixed the bug in the snow DA. Could we try again on Hera?

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve.

**Pull request merged**
